app/code/Magento/PageBuilder/view/frontend/templates/catalog/product/widget/content/carousel.phtml

```diff
@@ -61,6 +61,12 @@ use Magento\Framework\App\Action\Action;
                                         <?php if ($_item->isSaleable()): ?>
                                             <?php $postParams = $block->getAddToCartPostParams($_item); ?>
                                             <form data-role="tocart-form" data-product-sku="<?= $block->escapeHtml($_item->getSku()) ?>" action="<?= $block->escapeUrl($postParams['action']) ?>" method="post">
+                                                <?php $options = $block->getViewModel()->getOptionsData($_item); ?>
```

Comment:
My suggestion is  to change it on this way, `$options = $block->getViewModel()?->getOptionsData($_item)` to prevent nullable (not defined) viewModel.

Reply:
@Bashev Thanks for your suggestion. I've updated the code as per your suggestion.